src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -3596,10 +3604,34 @@ mongoc_collection_find_and_modify_with_opts (
       if (retry_server_stream && retry_server_stream->sd->max_wire_version >=
                                     WIRE_VERSION_RETRY_WRITES) {
          parts.assembled.server_stream = retry_server_stream;
+         {
+            // Store the original error and reply before retry.
+            BSON_ASSERT (!original_error.set); // Retry only happens once.
+            original_error.set = true;
+            bson_copy_to (reply_ptr, &original_error.reply);
+            if (error) {
+               original_error.error = *error;
+            }
+         }
          GOTO (retry);
       }
    }
 
+   // If a retry attempt fails with an error labeled NoWritesPerformed,
+   // drivers MUST return the original error.
+   if (original_error.set && !ret &&
```

Comment:
```suggestion
   if (original_error.set &&
```
Suggestion to add `!ret` was mistaken as `ok: 1` may be present alongside error labels.